### PR TITLE
Issue #1913 Adding unit test for GreaterOrEqualToBaseVersion

### LINKED ISSUE #1913: Add an unit test for GreaterOrEqualToBaseVersion()
I noticed that we do not have a go unit test for GreaterOrEqualToBaseVersion(). Hence going to add one.

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Fix #1913  
Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
CI failing because of unavailability of nip.io
```
07:42:19     Then with up to "10" retries with wait period of "500ms" the "status code" of HTTP request to "/" of service "ruby-ex" in namespace "ruby" is equal to "200"                                # integration_test.go:755 -> verifyRequestToServiceWithRetry
07:42:19     Server returned error on url: http://ruby-ex-ruby.192.168.42.193.nip.io/
```

File: pkg/minishift/openshift/version/openshift_versions_test.go [hunk 1]
```diff
@@ -133,3 +133,25 @@ func Test_validate_openshift_min_versions(t *testing.T) {
...
+	}
+
+	for _, versionTest := range versionTestData {
+		actualResult, _ := IsGreaterOrEqualToBaseVersion(versionTest.openshiftVersion, versionTest.baseOpenshiftVersion)
+		if actualResult != versionTest.expectedResult {
+			t.Errorf("IsGreaterOrEqualToBaseVersion(%s, %s) is expected to return '%v' but returned '%v'",
+				versionTest.openshiftVersion, versionTest.baseOpenshiftVersion, versionTest.expectedResult, actualResult)
+		}
+	}
+}
```
Comment (praveenkumar) [inline pkg/minishift/openshift/version/openshift_versions_test.go]: Test for GreaterOrEqualToBaseVersion already exist and also have good test table along with error handle https://github.com/minishift/minishift/blob/master/pkg/minishift/openshift/version/openshift_versions_test.go#L100-L135 

I think we can just close this PR and relevant issue.

Comment (LalatenduMohanty) [reply pkg/minishift/openshift/version/openshift_versions_test.go]: @praveenkumar Also as discussed I have removed https://github.com/minishift/minishift/blob/master/pkg/minishift/openshift/version/openshift_versions_test.go#L100-L135 as we have https://github.com/minishift/minishift/blob/master/pkg/minishift/openshift/version/openshift_versions_test.go#L30-L65

File: pkg/minishift/openshift/version/openshift_versions_test.go [hunk 2]
```diff
@@ -97,39 +97,34 @@ func TestPrintDownStreamVersions(t *testing.T) {
...
+		{"v3.6.0", "v3.7.0", false, nil},
+		{"v3.6.0-alpha.1", "v3.6.0-alpha.2", false, nil},
+		{"v3.7.1", "v3.7.1", true, nil},
+		{"v3.6.0-rc1", "v3.6.0-alpha.1", true, nil},
+		{"v3.6.0-alpha.1", "v3.6.0-beta.0", false, nil},
+		{"v1.4.1", "v1.4.1", true, nil},
+		{"v1.5.0-alpha.0", "v1.4.1", true, nil},
+		{"v1.5.1-beta.0", "v1.4.1", true, nil},
+		{"foo", "v1.4.1", false, errors.New("Invalid version format 'foo': No Major.Minor.Patch elements found")},
+		{"151", "v1.4.1", false, errors.New("Invalid version format '151': No Major.Minor.Patch elements found")},
```
Comment (praveenkumar) [inline pkg/minishift/openshift/version/openshift_versions_test.go]: 👍

File: pkg/minishift/openshift/version/openshift_versions_test.go [hunk 3]
```diff
@@ -97,39 +97,34 @@ func TestPrintDownStreamVersions(t *testing.T) {
...
-
-		if err != nil && err.Error() != versionTest.err.Error() {
-			t.Fatalf("Unexpected error. Expected '%v', got '%v'", versionTest.err, err)
-		}
-
-		if valid != versionTest.valid {
-			t.Fatalf("Expected '%t' Got '%t' for %s", versionTest.valid, valid, versionTest.version)
+		if actualErr != nil && actualErr.Error() != versionTest.expectedErr.Error() {
+			t.Errorf("IsGreaterOrEqualToBaseVersion(%s, %s) is expected to return '%v' but returned '%v'",
+				versionTest.openshiftVersion, versionTest.baseOpenshiftVersion, versionTest.expectedErr, actualErr)
```
Comment (praveenkumar) [inline pkg/minishift/openshift/version/openshift_versions_test.go]: 👍

**Comment by budhrg** [comment]
retest this please

**Comment by budhrg** [comment]
retest this please

**Comment by budhrg** [comment]
retest this please

**Pull request merged**